core/src/main/java/com/sequenceiq/cloudbreak/controller/StackCreatorService.java

```diff
@@ -379,9 +382,43 @@ void fillInstanceMetadata(DetailedEnvironmentResponse environment, Stack stack)
                 instanceMetaData.setPrivateId(privateIdNumber++);
                 instanceMetaData.setInstanceStatus(InstanceStatus.REQUESTED);
             }
-            multiAzCalculatorService.calculateByRoundRobin(subnetAzPairs, instanceGroup);
-            prepareInstanceMetaDataSubnetAndAvailabilityZoneAndRackId(stackSubnetId, stackAz, instanceGroup);
         }
+        if (stack.getType() == StackType.DATALAKE) {
+            for (InstanceGroup instanceGroup : sortInstanceGroups(stack)) {
+                if (!(instanceGroup.getGroupName().equals(InstanceGroupName.GATEWAY) || instanceGroup.getGroupName().equals(InstanceGroupName.AUXILIARY))) {
```

Comment:
why don't we transform this condition it would be more straight forward if it would be `!instanceGroup.getGroupName().equals(InstanceGroupName.GATEWAY) && !instanceGroup.getGroupName().equals(InstanceGroupName.AUXILIARY)`?
Why do we not want to run the calculating logic for gateway group? The previous logic applied the same logic on all of the groups. -> Shouldn't it be feasible only for avoiding the run of the logic in case of `AUXILIARY` groups?

Reply:
We are applying the calculating logic for all the groups. We would like to treat gateway and Auxiliary groups differently.
These host groups share services so we would like to get a combined list of InstanceMetaData and pass it to calculating logic.

Reply:
I meant the `multiAzCalculatorService.calculateByRoundRobin` is called for `AUXILIARY` explicitly at the end of the logic, but the the mentioned conditions avoid the mentioned calculation for instance group with name `GATEWAY`.
Can we get proper UT coverage for this logic please?